### ISSUE TITLE
Owncloud Admin panel states, "no working internet connection"

### ISSUE THREAD

**Issue by amc2002** [owncloud/core#19138]
I was trying to resolve this issue in the forum, and was informed I should open a bug for it. I am getting the notice in the admin panel, "This server has no working internet connection. This means that some of the features like mounting external storage, notifications about updates or installation of third-party apps will not work. Accessing files remotely and sending of notification emails might not work, either. We suggest to enable internet connection for this server if you want to have all features."

My server appears in the hosts file as 127.0.0.1 and as its own IP. I have also confirmed everything suggested in the FAQ for solving this issue (see numbered items below). Looking for other suggestions to help resolve this issue. Thanks.

1. "A broken, outgoing connection to the appstore (https://apps.owncloud.com)". The 443 port is opened, the *wget* example for apps.owncloud.com works correctly without issue (used wget, not ping).
2. "Buggy versions of OpenSSL/NSS (oC 8.1.1 should show this in the admin backend)." There is no notification in the admin backend.
3. "A missing config/ca-bundle.crt." A file with this name is present in <OC Folder>/config
4. "A disabled appstore ('appstoreenabled' => false, in your config.php)". There is no such string in config.php - see below.
5. "A wrong configured appstore URL (Remove 'appstoreurl' => from your config.php)." There is no such string in config.php 
6. "Empty or wrong configured 'proxy' => '', in config.php." There is no such string in config.php - see below

My config.php (private information noted by @@@@)

<?php
$CONFIG = array (
'instanceid' => '@@@@',
'passwordsalt' => '@@@@',
'secret' => '@@@@@@',
'trusted_domains' =>
array (
0 => 'owncloud.@@@@.com',
),
'datadirectory' => '/var/www/html/owncloud/data',
'overwrite.cli.url' => 'https://owncloud.@@@@.com/owncloud',
'dbtype' => 'mysql',
'version' => '8.1.3.0',
'dbname' => 'ownCloud',
'dbhost' => 'localhost',
'dbtableprefix' => 'oc_',
'dbuser' => '@@@@@@',
'dbpassword' => '@@@@@',
'installed' => true,
'ldapIgnoreNamingRules' => false,
'mail_from_address' => 'webmaster',
'mail_smtpmode' => 'smtp',
'mail_domain' => '@@@@.com',
'loglevel' => 2,
'forcessl' => true,
'forceSSLforSubdomains' => true,
'mail_smtpauthtype' => 'LOGIN',
'mail_smtpauth' => 1,
'mail_smtphost' => '@@@@.com',
'mail_smtpport' => '465',
'mail_smtpname' => 'sendmail@@@@@.com',
'mail_smtppassword' => '@@@@@',
'mail_smtpsecure' => 'ssl',
'maintenance' => false,
);

**Comment by karlitschek** [comment]
@LukasReschke any idea?

**Comment by LukasReschke** [comment]
=> https://github.com/owncloud/core/blob/master/CONTRIBUTING.md#short-version

**Comment by amc2002** [comment]
Hope this helps

### Server configuration
**Operating system**: CentOS 7 (up-to-date) Linux ownCloud 3.10.0-229.4.2.el7.x86_64
**Web server:**: Apache/2.4.6 (CentOS) OpenSSL/1.0.1e-fips 
**Database:**: mariadb-server-5.5.44-1.el7_1.x86_64
**PHP version:**: 5.4.16
**ownCloud version:** : 8.1.3 (stable)
**Updated from an older ownCloud or fresh install:** : Updated
**List of activated apps:** 
Enabled:
 - activity: 2.0.2
 - files: 1.1.10
 - files_external: 0.2.3
 - files_locking:
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_trashbin: 0.6.3
 - files_versions: 1.0.6
 - files_videoviewer: 0.1.3
 - firstrunwizard: 1.1
 - gallery: 0.6.0
 - provisioning_api: 0.2
 - templateeditor: 0.1
 - user_ldap: 0.6.1
Disabled:
 - encryption
 - external
 - user_external
 - user_webdavauth

**The content of config/config.php:** As above
**Are you using external storage: no
**Are you using encryption:** no
**Are you using an external user-backend, if yes which one:** LDAP

#### LDAP configuration (delete this part if not used)
| Configuration                 | s01                                                |
+-------------------------------+----------------------------------------------------+
| hasMemberOfFilterSupport      | 1                                                  |
| hasPagedResultSupport         |                                                    |
| homeFolderNamingRule          |                                                    |
| lastJpegPhotoLookup           | 0                                                  |
| ldapAgentName                 | CN=@@@@@@@@    |
| ldapAgentPassword             | ***                                                |
| ldapAttributesForGroupSearch  |                                                    |
| ldapAttributesForUserSearch   |                                                    |
| ldapBackupHost                |                                                    |
| ldapBackupPort                |                                                    |
| ldapBase                      | DC=@@@@                          |
| ldapBaseGroups                | DC=@@@@@                           |
| ldapBaseUsers                 | DC=@@@@@                           |
| ldapCacheTTL                  | 600                                                |
| ldapConfigurationActive       | 1                                                  |
| ldapEmailAttribute            | mail                                               |
| ldapExperiencedAdmin          | 0                                                  |
| ldapExpertUUIDGroupAttr       |                                                    |
| ldapExpertUUIDUserAttr        |                                                    |
| ldapExpertUsernameAttr        |                                                    |
| ldapGroupDisplayName          | cn                                                 |
| ldapGroupFilter               |                                                    |
| ldapGroupFilterGroups         |                                                    |
| ldapGroupFilterMode           | 0                                                  |
| ldapGroupFilterObjectclass    |                                                    |
| ldapGroupMemberAssocAttr      | uniqueMember                                       |
| ldapHost                      | @@@@@                                    |
| ldapIgnoreNamingRules         |                                                    |
| ldapLoginFilter               | (&(&(|(objectclass=person)))(samaccountname=%uid)) |
| ldapLoginFilterAttributes     |                                                    |
| ldapLoginFilterEmail          | 0                                                  |
| ldapLoginFilterMode           | 1                                                  |
| ldapLoginFilterUsername       | 1                                                  |
| ldapNestedGroups              | 0                                                  |
| ldapNoCase                    | 0                                                  |
| ldapOverrideMainServer        | 0                                                  |
| ldapPagingSize                | 500                                                |
| ldapPort                      | 389                                                |
| ldapQuotaAttribute            |                                                    |
| ldapQuotaDefault              |                                                    |
| ldapTLS                       | 0                                                  |
| ldapUserDisplayName           | displayname                                        |
| ldapUserFilter                | (&(|(objectclass=person)))                         |
| ldapUserFilterGroups          |                                                    |
| ldapUserFilterMode            | 1                                                  |
| ldapUserFilterObjectclass     | person                                             |
| ldapUuidGroupAttribute        | auto                                               |
| ldapUuidUserAttribute         | auto                                               |
| turnOffCertCheck              | 0                                                  |
| useMemberOfToDetectMembership | 1                          

### Logs
#### Web server error log
The only entry here is the following recurrent message:
[Thu Sep 17 12:16:39.113289 2015] [authz_core:error] [pid 12237] [client 1.2.3.4:41285] AH01630: client denied by server configuration: /var/www/html/owncloud/data/htaccesstest.txt
[Thu Sep 17 12:53:11.052328 2015] [authz_core:error] [pid 24101] [client 1.2.3.4:42645] AH01630: client denied by server configuration: /var/www/html/owncloud/data/htaccesstest.txt
[Thu Sep 17 13:01:27.483510 2015] [authz_core:error] [pid 28356] [client 1.2.3.4:42940] AH01630: client denied by server configuration: /var/www/html/owncloud/data/htaccesstest.txt

Insert your webserver log here
only entries in access_log are recurring client notifications:
[17/Sep/2015:13:18:47 -0400] "PROPFIND /owncloud/remote.php/webdav/ HTTP/1.1" 207 416
only entries in request_log are recurring client notifications:
TLSv1.2 ECDHE-RSA-AES128-GCM-SHA256 "PROPFIND /owncloud/remote.php/webdav/ HTTP/1.1" 1151

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
Info	user_ldap	No group filter is specified, LDAP group feature will not be used.	2015-09-17T17:16:43+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T17:16:22+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T17:09:22+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T17:01:27+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T16:53:10+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T16:16:38+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T16:16:25+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T16:10:19+00:00
Error	PHP	Undefined offset: 0 at /var/www/html/owncloud/lib/private/appframework/http/request.php#685	2015-09-17T16:09:48+00:00
Error	PHP	Undefined offset: 0 at /var/www/html/owncloud/lib/private/appframework/http/request.php#685	2015-09-17T16:09:48+00:00
Error	PHP	fopen(/var/www/html/owncloud/config/config.php): failed to open stream: No such file or directory at /var/www/html/owncloud/lib/private/config.php#191	2015-09-17T16:09:29+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T15:40:18+00:00
Error	PHP	ldap_read(): Search: Invalid DN syntax at /var/www/html/owncloud/apps/user_ldap/lib/ldap.php#257	2015-09-17T15:22:48+00:00
Debug	user_ldap	initializing paged search for Filter objectClass=* base Array ( [0] => @@@@@@@@@@@@@@@@@@ ) attr Array ( [0] => ) limit 500 offset 0	2015-09-17T17:17:01+00:00
Info	user_ldap	No group filter is specified, LDAP group feature will not be used.	2015-09-17T17:17:01+00:00
Debug	user_ldap	readAttribute: @@@@@@@@@@@@@ found	2015-09-17T17:17:00+00:00
Debug	user_ldap	Ready for a paged search	2015-09-17T17:17:00+00:00
Debug	user_ldap	initializing paged search for Filter objectClass=* base Array ( [0] => @@@@@@@@@@@@@@ ) attr Array ( [0] => ) limit 500 offset 0	2015-09-17T17:17:00+00:00
Debug	user_ldap	Ready for a paged search	2015-09-17T17:17:00+00:00
Debug	user_ldap	initializing paged search for Filter objectClass=* base Array ( [0] => @@@@@@@@@@@@@@@ ) attr Array ( [0] => displayname ) limit 500 offset 0	2015-09-17T17:17:00+00:00
Debug	user_ldap	readAttribute: @@@@@@@@@@@@ found	2015-09-17T17:17:00+00:00
Debug	user_ldap	Ready for a paged search	2015-09-17T17:17:00+00:00
Debug	user_ldap	initializing paged search for Filter objectClass=* base Array ( [0] => @@@@@@@@@@@@@@@@@@@ ) attr Array ( [0] => ) limit 500 offset 0	2015-09-17T17:17:00+00:00

**Comment by amc2002** [comment]
After doing some more digging, I think I'm getting hit with this: https://github.com/owncloud/core/issues/16255

**Comment by RealRancor** [comment]
@amc2002 Yes, the "No internet connection" is caused by the "cURL is using an outdated NSS version".

You had stated in your forums post that you didn't got an message about the NSS in your admin backend. But maybe i should describe more clearly in the FAQ that i mean the cURL message with "buggy OpenSSL/NSS versions".

**Comment by amc2002** [comment]
Got it. Yeah, my bad, because the NSS message was actually there, I was just looking for it next to the "no internet connection" since I figured if they were related, they would be together. 

I would say for me, anyway, the two being related wasn't necessarily clear until I did some more digging through previous threads here. :) 

I think perhaps it's just the error messages on the security page that could use some tweaking, especially since one of the bug reports for cURL notes that this has been going on for a year, at least under RedHat, so I have little confidence the CentOS implementation will be updated anytime soon. 

I will go back and update my forum post and mention that the two messages are related.

Thanks for the help, guys. I'll ignore the messages for now, and if they annoy me at some point I'll try updating curl by hand using the suggested method on the other page.
